### PR TITLE
add Go mod support

### PR DESCRIPTION
**Pull request by marten-seemann**

### CODE REVIEW

**Comment by anacrolix** [comment]
I don't think it's correct to use `make deps` for the go mod case. There's a lot of activity around the "default" travis file at the moment. For your case with ginkgo, it may be necessary to unpack some of the contents of the magical bash script that gets imported. https://github.com/libp2p/go-libp2p-kad-dht/pull/279/files.

**Comment by anacrolix** [comment]
Okay I see your update, I'll have another look when the CI finishes.

**Comment by marten-seemann** [comment]
@anacrolix I requested your review too early, before making sure that the CI passes. Sorry for that!
I'm playing around with the Travis config now, I'll re-request a review once everything passes.

**Comment by marten-seemann** [comment]
@anacrolix Travis seems happy now. Can you please have another look?

File: .travis.yml [hunk 1]
```diff
@@ -7,15 +7,26 @@ go:
...
-  - make deps
+  - |
+    if [ ${BUILDMODE} == "gx" ]; then
+      make deps;
+    fi
+  - |
+    if [ ${BUILDMODE} == "gomod" ]; then
+      export GO111MODULE=on;
+      make testutils;
+      go get .;
```
Comment (anacrolix) [inline .travis.yml]: What's the go get for?

**Review by anacrolix — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (marten-seemann) [reply .travis.yml]: Good question. Seems to work without it. I removed it.

**Pull request merged**